xmake/modules/package/tools/cmake.lua

```diff
@@ -755,6 +723,95 @@ function _get_configs_for_install(package, configs, opt)
+            cmake_default_flags.cflags = cmake_default_flags.cflags .. " " .. outdata:match(format("CMAKE_C_FLAGS_%s is (.-)\n", buildtype)):replace("/MDd", ""):replace("/MD", "")
+            cmake_default_flags.cxxflags = outdata:match("CMAKE_CXX_FLAGS is (.-)\n") or " "
+            cmake_default_flags.cxxflags = cmake_default_flags.cxxflags .. " " .. outdata:match(format("CMAKE_CXX_FLAGS_%s is (.-)\n", buildtype)):replace("/MDd", ""):replace("/MD", "")
+            cmake_default_flags.ldflags = outdata:match("CMAKE_EXE_LINKER_FLAGS is (.-)\n") or " "
+            cmake_default_flags.ldflags = cmake_default_flags.ldflags .. " " .. outdata:match(format("CMAKE_EXE_LINKER_FLAGS_%s is (.-)\n", buildtype))
+            cmake_default_flags.shflags = outdata:match("CMAKE_SHARED_LINKER_FLAGS is (.-)\n") or " "
+            cmake_default_flags.shflags = cmake_default_flags.shflags .. " " .. outdata:match(format("CMAKE_SHARED_LINKER_FLAGS_%s is (.-)\n", buildtype))
+            cmake_default_flags.arflags = outdata:match("CMAKE_STATIC_LINKER_FLAGS is (.-)\n") or " "
+            cmake_default_flags.arflags = cmake_default_flags.arflags .. " " ..outdata:match(format("CMAKE_STATIC_LINKER_FLAGS_%s is (.-)\n", buildtype))
+
+            _g.cmake_default_flags = _g.cmake_default_flags or {}
+            _g.cmake_default_flags[buildtype] = cmake_default_flags
+        end
+        os.rm(tmpdir)
+    end
+    return cmake_default_flags
+end
+
+function _get_cmake_buildtype(package)
+    local cmake_buildtype_map = {
+        debug = "DEBUG",
+        release = "RELEASE",
+        releasedbg = "RELWITHDEBINFO"
+    }
+    local buildtype = package:mode()
+    return cmake_buildtype_map[buildtype] or "RELEASE"
+end
+
+function _get_envs_for_default_flags(package, configs, opt)
+    local buildtype = _get_cmake_buildtype(package)
+    local envs = {}
+    local default_flags = _get_default_flags(package, configs, buildtype, opt)
+    if default_flags then
+        envs[format("CMAKE_CXX_FLAGS_%s", buildtype)]           = (not opt.cxxflags and not opt.cxflags) and default_flags.cxxflags
+        envs[format("CMAKE_C_FLAGS_%s", buildtype)]             = (not opt.cflags and not opt.cxflags) and default_flags.cflags
+        envs[format("CMAKE_EXE_LINKER_FLAGS_%s", buildtype)]    = not opt.ldflags and default_flags.ldflags
+        envs[format("CMAKE_STATIC_LINKER_FLAGS_%s", buildtype)] = not opt.arflags and default_flags.arflags
+        envs[format("CMAKE_SHARED_LINKER_FLAGS_%s", buildtype)] = not opt.shflags and default_flags.shflags
+    end
+    return envs
+end
+
+function _get_envs_for_runtime_flags(package, configs, opt)
+    local buildtype = _get_cmake_buildtype(package)
+    local envs = {}
+    local runtimes = package:runtimes()
+    if runtimes then
+        local fake_target = {is_shared = function(_) return false end,
+                             sourcekinds = function(_) return "c" end}
```

Comment:
why not use package? it will break if package toolchain is changed. 

https://github.com/xmake-io/xmake/issues/5542